addons/web/static/src/core/popover/popover_service.js

```diff
@@ -118,13 +129,35 @@ export const popoverService = {
             /**
              * Signals the manager to remove the popover with key = `key`.
              *
-             * @param {string} key
+             * @param {any} key
              */
             remove(key) {
                 bus.trigger("REMOVE", key);
             },
         };
     },
+    specializeForComponent(component, service) {
+        const serviceAdd = service.add;
+        return Object.assign({}, service, {
+            add(params) {
+                if (!("key" in params)) {
+                    params.key = Symbol();
+                }
+                const paramsOnClose = params.onClose;
+                if (params.onClose) {
+                    params.onClose = function () {
+                        if (component.__owl__.status !== 5 /* DESTROYED */) {
+                            paramsOnClose();
+                        }
+                    };
+                }
+                onWillUnmount(() => {
```

Comment:
this is actually wrong. the add method may be called at any time, and onwillunmount will then be wrongly binded to another component.